pvlib/pvsystem.py

```diff
@@ -285,24 +350,19 @@ def get_irradiance(self, solar_zenith, solar_azimuth, dni, ghi, dhi,
         poa_irradiance : DataFrame
             Column names are: ``total, beam, sky, ground``.
         """
+        dni = self._validate_per_array(dni, system_wide=True)
+        ghi = self._validate_per_array(ghi, system_wide=True)
+        dhi = self._validate_per_array(dhi, system_wide=True)
```

Comment:
This precludes the use case where e.g. `ghi` is specified by array. Commenting to have a discussion if we want to accommodate this use case, or defer to a later PR if the capability becomes important.

Reply:
You're suggesting passing irradiance for each `Array` as a single ndarray as for #1085? I'm inclined to stick with a single acceptable input type (`tuple`) at this point, and discuss adding support for arrays in the future.

Reply:
Thinking more about the use case where more than one weather station is associated with a PVSystem, those system are almost certainly large enough to have many inverters, and so would be modeled by many instances of `PVSystem`. I'm OK with one `ghi` time series per `PVSystem`.